aiida/backends/tests/orm/implementation/test_comments.py

```diff
@@ -98,3 +116,299 @@ def test_creation_with_time(self):
+        # Create comment
+        comment = self.create_comment()
+        comment.store()
+        comment_uuid = str(comment.uuid)
+
+        # Make sure it exists
+        builder = orm.QueryBuilder().append(orm.Comment, project='uuid')
+        no_of_comments = builder.count()
+        found_comments_uuid = [_[0] for _ in builder.all()]
+        self.assertIn(comment_uuid, found_comments_uuid)
+
+        # Delete the comment
+        self.backend.comments.delete(comment.id)
+
+        # Check it was deleted
+        builder = orm.QueryBuilder().append(orm.Comment, project='uuid')
+        self.assertEqual(builder.count(), no_of_comments - 1)
+        found_comments_uuid = [_[0] for _ in builder.all()]
+        self.assertNotIn(comment_uuid, found_comments_uuid)
+
+    def test_delete_all(self):
+        """Test `delete_all` method"""
+        # Create comment
+        self.create_comment().store()
+
+        # Make sure there are Comments in the db
+        self.assertGreater(len(orm.Comment.objects.all()), 0)
+
+        # Delete all Comments
+        self.backend.comments.delete_all()
+
+        # Check they were deleted
+        self.assertEqual(len(orm.Comment.objects.all()), 0)
+
+    def test_delete_many_no_filters(self):
+        """Test `delete_many` method with empty filters"""
+        # Create comment
+        self.create_comment().store()
+
+        # Make sure there are Comments in the db
+        count = len(orm.Comment.objects.all())
+        self.assertGreater(count, 0)
+
+        # Pass empty filter to delete_many, making sure ValidationError is raised
+        with self.assertRaises(exceptions.ValidationError):
+            self.backend.comments.delete_many({})
+
+        # Check nothing was deleted
+        self.assertEqual(len(orm.Comment.objects.all()), count)
+
+    def test_delete_many_ids(self):
+        """Test `delete_many` method filtering on both `id` and `uuid`"""
+        # Create comments
+        comment1 = self.create_comment()
+        comment2 = self.create_comment()
+        comment3 = self.create_comment()
+        comment_uuids = []
+        for comment in [comment1, comment2, comment3]:
+            comment.store()
+            comment_uuids.append(str(comment.uuid))
+
+        # Make sure they exist
+        builder = orm.QueryBuilder().append(orm.Comment, project='uuid')
+        no_of_comments = builder.count()
+        self.assertGreater(no_of_comments, 0)
+        found_comments_uuid = [_[0] for _ in builder.all()]
+        for comment_uuid in comment_uuids:
+            self.assertIn(comment_uuid, found_comments_uuid)
+
+        # Delete last two comments (comment2, comment3)
+        filters = {'or': [{'id': comment2.id}, {'uuid': str(comment3.uuid)}]}
+        self.backend.comments.delete_many(filters=filters)
+
+        # Check they were deleted
+        builder = orm.QueryBuilder().append(orm.Comment, project='uuid')
+        self.assertEqual(builder.count(), no_of_comments - 2)
```

Comment:
Not exactly, since I am not deleting all Comments between tests (to avoid using `delete_all()` outside of a class that tests this exact method). However, I have rewritten it using filters for the QueryBuilder, making it possible to do this.